bookstore/clone.py

```diff
@@ -40,6 +37,68 @@ class BookstoreCloneHandler(IPythonHandler):
     `Jupyter Notebook reference on Custom Handlers <https://jupyter-notebook.readthedocs.io/en/stable/extending/handlers.html#registering-custom-handlers>`_
     """
 
+    def initialize(self):
+        """Helper to retrieve bookstore setting for the session."""
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+
+    @web.authenticated
+    async def get(self):
+        """GET /bookstore/clone?s3_bucket=<your_s3_bucket>&s3_key=<your_s3_key>
+
+        Renders an options page that will allow you to clone a notebook
+        from a specific bucket via the Bookstore cloning API.
+        """
+        s3_bucket = self.get_argument("s3_bucket")
+        if s3_bucket == '' or s3_bucket == "/":
+            raise web.HTTPError(400, "Must have a bucket to clone from")
+
+        # s3_paths module has an s3_key function; s3_object_key avoids confusion
+        s3_object_key = self.get_argument("s3_key")
+        if s3_object_key == '' or s3_object_key == '/':
+            raise web.HTTPError(400, "Must have a key to clone from")
```

Comment:
```suggestion
            raise web.HTTPError(400, "Requires an S3 key")
```

Reply:
I feel like conveying that this is functionality related to cloning would also be useful here, what is the reason for removing that info? Other than the loss of the cloning specific info, the rephrasing improves clarity

Reply:
Same as above. Add clone to the end of the message.